cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -136,6 +129,27 @@ type API struct {
 	BindPort int32
 }
 
+// NodeRegistrationOptions holds fields that relate to registering a new master or node to the cluster, either via "kubeadm init" or "kubeadm join"
+type NodeRegistrationOptions struct {
```

Comment:
If we are considering convergence towards cluster API, the [corresponding object](https://github.com/kubernetes/kube-deploy/blob/5d7e0ecaec541c561f6e18cb4500c11554cfa406/cluster-api/pkg/apis/cluster/v1alpha1/machine_types.go#L51) is named `machineSpec`. 
WDYT? it is too early for starting to unify naming?